src/gui/DatabaseOpenWidget.cpp

```diff
@@ -512,7 +511,10 @@ void DatabaseOpenWidget::pollHardwareKey(bool manualTrigger)
     m_pollingHardwareKey = true;
     m_manualHardwareKeyRefresh = manualTrigger;
 
-    YubiKey::instance()->findValidKeysAsync();
+    // Add a delay, if this is an automatic trigger, to allow the USB device to settle as
+    // the device may not report a valid serial number immediately after plugging in
+    int delay = manualTrigger ? 0 : 1000;
```

Comment:
I think 1000 is way too much. It adds visible delay, which will cause confusion and frustration.

Reply:
Disagree, I tested this and saw no difference in delay. The polling itself takes several seconds.

Reply:
But it will add a flat 1000ms on top before it even starts. I would prefer to reduce it to 500ms.